src/components/SiteTrackingRequired/SiteTrackingRequired.test.js

```diff
@@ -0,0 +1,110 @@
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.upgrade_account_free_title')).toBeInTheDocument();
+  });
+
+  it('should show trial not accepted messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.allow_enable_trial_title')).toBeInTheDocument();
+  });
+
+  it('should show not domains messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.thereAreNotDomains;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.datahub_not_domains_title')).toBeInTheDocument();
+  });
+
+  it('should be activate the trial and redirect to control panel site tracking settings', async () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => true,
+      },
+      appConfiguration: { dopplerLegacyUrl: 'localhost:3000' },
+      window: { location: { href: '/ControlPanel/CampaignsPreferences/SiteTrackingSettings' } },
+    };
+
+    // Act
+    const { getByText, container } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.allow_enable_trial_title')).toBeInTheDocument();
+    expect(dependencies.window.location.href).toEqual(
+      '/ControlPanel/CampaignsPreferences/SiteTrackingSettings',
+    );
+
+    container.querySelector('button').click();
+    await waitForDomChange();
+
+    expect(dependencies.window.location.href).toEqual(
+      'localhost:3000/ControlPanel/CampaignsPreferences/SiteTrackingSettings',
```

Comment:
Are you sure this is ok? you have in line 99 a relative url, 
``` expect(dependencies.window.location.href).toEqual(
      '/ControlPanel/CampaignsPreferences/SiteTrackingSettings',
    );
```
and here a complete url.